ibis/backends/bigquery/__init__.py

```diff
@@ -395,14 +405,19 @@ def to_pyarrow_batches(
         chunk_size: int = 1_000_000,
         **kwargs: Any,
     ):
-        self._import_pyarrow()
+        pa = self._import_pyarrow()
+
+        schema = expr.as_table().schema()
 
-        # kind of pointless, but it'll work if there's enough memory
         query_ast = self.compiler.to_ast_ensure_limit(expr, limit, params=params)
         sql = query_ast.compile()
         cursor = self.raw_sql(sql, params=params, **kwargs)
-        table = self._cursor_to_arrow(cursor)
-        return table.to_reader(chunk_size)
+        batch_iter = self._cursor_to_arrow(
+            cursor,
+            method=lambda result: result.to_arrow_iterable(),
```

Comment:
Does the iterator returned by `to_arrow_iterable` hold references to the cursor/connection/whatever needed to keep things working, or do we need to do the `weakref` trick we did for `duckdb` here too?

Reply:
I'm not sure if the model maps onto the same kind of closable cursor that the sqlalchemy backends have.

Reply:
@tswast Should we be judicious about cleaning up the cursor here?

Reply:
It looks like the bigquery cursor is purely a compatibility layer.

This iterator is an http iterator, whose connection state semantics I don't fully understand.

Reply:
It looks like `HTTPIterator` has no connection state, and is making a get request per page of results.

Reply:
As far as cleanup goes, my hope is that when the BQ Storage client goes out-of-scope it should clean itself up, but there might be some open issues there in the BigQuery client.

Reply:
Searching around the Storage Client issue tracker I couldn't find any connection issues. Let's leave it as is for now.